### ISSUE TITLE
[Bug Report] IoT Device Client - MethodRequest incorrectly implements toJson, makes module method calls fail

### ISSUE THREAD

**Issue by derSchtefan** [Azure/azure-iot-sdk-java#1460]
# Context

- **SDK version used:** iot-device-client:1.34.1

## Description of the issue
Package `com.microsoft.azure.sdk.iot.device.edge`,  `MethodRequest.java`, lines 37, 45, 59, and 80.
`payload` is incorrectly defined as String instead of Object. The toJson method of this class is called during execution and GSON will then encode the String again as JSON string . This makes it impossible to execute any module method calls with the Module Java SDK, unless it uses no payload or is designed to work with this weird quirk. 

Any target module out there just will receive double encoded junk and won't work. A direct call it from IoT hub (or using another SDK for another language) will work correctly.

## Code sample exhibiting the issue
```java
var device = moduleClient.getConfig().getDeviceId();

var body = Map.of("param1", "value1"); // this would usually be new MyRequestModel(.....)
var bodyAsString = (new GsonBuilder()).create().toJson(body);

var request = new MethodRequest("someMethod", bodyAsString); // ctor parameter and payload field are both "String". 
var response = moduleClient.invokeMethod(device, module, request);
```

will send a request of 

```json
{
    "methodName": "someMethod",
    "payload": "{\"param1\":\"value1\"}"
}
```

instead of 
```json
{
    "methodName": "someMethod",
    "payload": {
        "param1": "value1"
    }
}
```

## Suggested change
Make the payload of type Object. This will allow GSON to correctly serialize the MethodRequest. Any existing code out there (if there even is any) will continue to work correctly (since a String input serializes to the same String).

## Suggested workaround
While Microsoft is working on this issue, if anybody else stumbles over this, create this class that extends from MethodRequest and pass it as parameter to `invokeMethod`. I tested this with several 3rd party target modules.

```java
public class CorrectMethodRequest extends MethodRequest {
    private final Map<String,Object> requestMap;

    public CorrectMethodRequest(String methodName, Object payloadObject) {
        // just so this does not look odd in  debugger
        super(methodName, null);

        var map = new HashMap<String, Object>();
        map.put("methodName", methodName);
        map.put("payload", payloadObject);
        this.requestMap = map;
    }


    public CorrectMethodRequest(String methodName, Object payloadObject, Integer responseTimeout, Integer connectionTimeout) {
        // just so this does not look odd in  debugger
        super(methodName, null, responseTimeout, connectionTimeout);

        var map = new HashMap<String, Object>();
        map.put("methodName", methodName);
        map.put("payload", payloadObject);
        map.put("responseTimeoutInSeconds", responseTimeout);
        map.put("connectionTimeoutInSeconds", connectionTimeout);
        this.requestMap = map;
    }

    @Override
    public String toJson() {
        return (new GsonBuilder()).create().toJson(requestMap);
    }
}

/// 

var device = moduleClient.getConfig().getDeviceId();

var body = Map.of("param1", "value1"); // this would usually be new MyRequestModel(.....)

var request = new CorrectMethodRequest(method, body);
var response = moduleClient.invokeMethod(device, module, request);
```

**Comment by brycewang-microsoft** [comment]
Hi @derSchtefan, thanks for reporting us this issue with reproducing steps and suggested changes. We agree that `String payload` can cause some problems here. We plan to bring this change into our upcoming 2.X.X versions of the clients since it could be a breaking change for customers.

**Comment by timtay-microsoft** [comment]
The newly released Device Client version 2.0.0 has the fix for this issue, so I'm closing this thread.
